--- a/src/command/activate.rs
+++ b/src/command/activate.rs
@@ -586,6 +586,10 @@
             o.try_for_each(|en| {
                 let d =
                     en.wrap_err_with(|| eyre!("Failed to enter generation directory subdir"))?;
+                #[cfg(target_os = "macos")]
+                if d.file_name() == DARWIN_ANCHOR_NAME {
+                    return Ok(());
+                }
                 match str::parse::<usize>(d.file_name().to_string_lossy().to_string().as_str()) {
                     Err(e) => Err(eyre!("Failed to parse generation: {e}")),
                     result::Result::Ok(res) => {
